spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -246,12 +247,24 @@ private DeleteOrphanFiles.Result doExecute() {
     List<String> orphanFiles =
         findOrphanFiles(spark(), actualFileIdentDS, validFileIdentDS, prefixMismatchMode);
 
-    Tasks.foreach(orphanFiles)
-        .noRetry()
-        .executeWith(deleteExecutorService)
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
-        .run(deleteFunc::accept);
+    if (deleteFunc == null && table.io() instanceof SupportsBulkOperations) {
+      ((SupportsBulkOperations) table.io()).deleteFiles(orphanFiles);
```

Comment:
Unlike non-bulk deletes, this may fail the entire action if there is a bulk delete exception. I think we will have to add a helper method like this or similar?

```
private void deleteFiles(SupportsBulkOperations io, List<String> paths) {
  try {
    io.deleteFiles(paths);
    LOG.info("Deleted {} files using bulk deletes", paths.size());

  } catch (BulkDeletionFailureException e) {
    int deletedFilesCount = paths.size() - e.numberFailedObjects();
    LOG.warn(
        "Deleted only {} of {} files using bulk deletes",
        deletedFilesCount,
        paths.size());
  }
}
```

Reply:
Logging the number of removed files like above may help debugging.